src/runtime/slice.go

```diff
@@ -47,7 +47,13 @@ func sliceGrow(oldBuf unsafe.Pointer, oldLen, oldCap, newCap, elemSize uintptr)
 	// memory allocators, this causes some difficult to debug issues.
 	newCap = 1 << bits.Len(uint(newCap))
 
-	buf := alloc(newCap*elemSize, nil)
+	var layout unsafe.Pointer
+	// less type info here; can only go off element size
+	if elemSize < unsafe.Sizeof(uintptr(0)) {
+		layout = gcLayoutNoPtrs
+	}
```

Comment:
This is probably best solved by a small compiler change that will pass the layout to `runtime.sliceAppend` (like it does today with `runtime.alloc`).